schema/policy.go

```diff
@@ -671,25 +741,32 @@ func (policy *Policy) GetOtherResourceCondition() *ResourceCondition {
 	return policy.otherResourceCondition
 }
 
-func addCustomFilters(f map[string]interface{}, tenantId string, conditionFilters *conditionFilter) {
+func addCustomFilters(f map[string]interface{}, auth Authorization, conditionFilters *conditionFilter) {
 	if conditionFilters == nil {
 		return
 	}
 	filters := make([]map[string]interface{}, 0)
 	if conditionFilters.isOwner {
-		filters = append(filters, map[string]interface{}{"property": "tenant_id", "type": "eq", "value": tenantId})
+		switch auth.ScopingType() {
+		case ScopedToTenant:
```

Comment:
design issue: `if`s (or this `switch`) on `ScopedToTenant` are present in several places. instead of `if`ing, could we take advantage of polymorphism or decorate Domain scoped authorization with additional Tenant scope features?